HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,244 @@
+        std::cout << rotationMatrixFromRollPitchYaw << std::endl;
+    }
+
+    // Rotation Vector to Axis-angle
+    std::cout << "\nConverting Rotation Vector to Axis-Angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle(representations.rotationVector.norm(),
+                                      representations.rotationVector.normalized());
+    std::cout << "Axis:\n" << axisAngle.axis() << std::endl;
+    std::cout << "Angle:\n" << axisAngle.angle() << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting Axis-Angle to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(axisAngle);
+    std::cout << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting Quaternion to Rotation Matrix:" << std::endl;
+    auto transformationMatrixFromQuaternion = Eigen::Affine3d(transformationMatrix);
+    transformationMatrixFromQuaternion.linear() = quaternion.toRotationMatrix();
+    std::cout << transformationMatrixFromQuaternion.linear() << std::endl;
+
+    return transformationMatrixFromQuaternion;
+}
+
+std::string toString(RotationConvention convention)
+{
+    switch(convention)
+    {
+        case RotationConvention::XYZ_Intrinsic: return "XYZ_Intrinsic";
+        case RotationConvention::XYZ_Extrinsic: return "XYZ_Extrinsic";
+        case RotationConvention::ZYX_Intrinsic: return "ZYX_Intrinsic";
+        case RotationConvention::ZYX_Extrinsic: return "ZYX_Extrinsic";
+        default: throw std::invalid_argument("Invalid RotationConvention");
+    }
+}
+
+// The following function converts Rotation Matrix to Roll-Pitch-Yaw angles in radians.
+// The rotation convention we use here is that Roll is a rotation about x-axis,
+// Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+// The array is ordered by Ox, Oy and Oz, where Oa is the rotation about axis a.
+Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d &rotationMatrix, const RotationConvention &rotation)
+{
+    switch(rotation)
+    {
+        case RotationConvention::XYZ_Intrinsic: return rotationMatrix.eulerAngles(0, 1, 2);
+        case RotationConvention::XYZ_Extrinsic: return rotationMatrix.eulerAngles(2, 1, 0).reverse();
+        case RotationConvention::ZYX_Intrinsic: return rotationMatrix.eulerAngles(2, 1, 0).reverse();
+        case RotationConvention::ZYX_Extrinsic: return rotationMatrix.eulerAngles(0, 1, 2);
+        default: throw std::invalid_argument("Invalid rotation");
+    }
+}
+
+// The following function converts Roll-Pitch-Yaw angles in radians to Rotation Matrix.
+// This function takes an array of roll, pitch and yaw angles, and a rotation convention, as input parameters.
+// For Roll-Pitch-Yaw we define that roll is a rotation about x-axis, pitch is a rotation about y-axis
+// and yaw is a rotation about z-axis.
+// Array is ordered by Ox, Oy, Oz, where Oa is a rotation about the a-axis
```

Comment:
Should we add at the end ', where a = {x, y, z}' or ', where a is x, y, or z' or similar ? I remember Rune was not clear what a stands for.

Also, maybe use the same sentance form between this and line 206.

Reply:
Maybe replace `a` with `i`, which is a more recognized iterator?

Reply:
I believe Rune's point was that it doesn't matter that you understand whether its x, y or z. The point is that (any)-axis is unclear without intrinsic or extrinsic. What we have to make clear is that the order applies regardless of whether the rotations happen with static or moving axes, and that this is the reason extrinsic/intrinsic is another parameter..

Reply:
Aha, if that was the case then it doesn't matter. For me Oa or Oi or On is all the same. Up to you.